ethereum/contracts/bridgehub/bridgehub-deps/Registry.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.6.0) (proxy/Proxy.sol)
+
+pragma solidity ^0.8.13;
+
+import "./BridgehubBase.sol";
+import "../bridgehub-interfaces/IRegistry.sol";
+import "../../common/libraries/UncheckedMath.sol";
+import "../../common/libraries/Diamond.sol";
+
+contract BridgehubRegistryFacet is IRegistry, BridgehubBase {
+    using UncheckedMath for uint256;
+    string public constant override getName = "BridgehubRegistryFacet";
+
+    /// @notice Proof system can be any contract with the appropriate interface, functionality
+    function newStateTransition(address _stateTransition) external onlyGovernor {
+        // KL todo add checks here
+        require(!bridgehubStorage.stateTransitionIsRegistered[_stateTransition], "r35");
+        bridgehubStorage.stateTransitionIsRegistered[_stateTransition] = true;
+    }
+
+    /// @notice
+    // KL todo make _chainId
+    function newChain(
```

Comment:
What is the long term vision on how these methods should work?

Right now what I see:

1. The governor assigns `chainId` some `stateTransition`.
2. The governor calls this `stateTransition` so it can call the `setStateTransitionChainContract`.

It seems to me a bit dependent on a single entity. It is okay in the short term, but interesting how you see the long-term vision for this process, so we can make sure we are forward compatible with it. 

A thing that makes me uncomfortable is that we have two mappings:
`chainId => stateTransitionChainAddress`, one is on the bridgehub (the main one) and one on the `StateTransition` contract. These two mappings may get out of sync very easily

Reply:
I agree currently it is dependent on a single entity. The long term vision is permissionlessless. With ZK IP anybody will be register StateTransitions permissionlessly, and the funds will be safe. Then anybody can apply for a chainId at the Bridgehub, specifying the desired ST. Then ST launches a STC.

Reply:
@kelemeno I agree

Reply:
added